File: tests/e2e/fixtures/java/MultiThreadProgram.java
/**
 * Multi-threaded Java program for E2E testing
 * Tests thread management and concurrent debugging
 */
public class MultiThreadProgram {
    public static void main(String[] args) throws Exception {
        System.out.println("Starting multi-thread program");

        Thread t1 = new Thread(() -> {
            for (int i = 0; i < 10; i++) {
                System.out.println("Thread 1: " + i);
                try {
                    Thread.sleep(100);
                } catch (InterruptedException e) {
                    break;
                }
            }
        }, "WorkerThread-1");

        Thread t2 = new Thread(() -> {
            for (int i = 0; i < 10; i++) {
                System.out.println("Thread 2: " + i);
                try {
                    Thread.sleep(100);
                } catch (InterruptedException e) {
                    break;
                }
            }
        }, "WorkerThread-2");

        Thread t3 = new Thread(() -> {
            for (int i = 0; i < 5; i++) {
                System.out.println("Thread 3: " + i);
                try {
                    Thread.sleep(200);
                } catch (InterruptedException e) {
                    break;
                }
            }
        }, "WorkerThread-3");

        // Start threads
        t1.start();
        t2.start();
        t3.start();

        // Wait for threads to complete
        t1.join();
        t2.join();
        t3.join();

        System.out.println("All threads completed");
    }
}

File: tests/e2e/fixtures/go/multi_thread_program.go
// Multi-thread program for E2E testing
package main

import (
	"fmt"
	"sync"
	"time"
)

// worker is a goroutine worker function
func worker(id int, wg *sync.WaitGroup, results chan<- int) {
	defer wg.Done()

	fmt.Printf("Worker %d starting\n", id)
	time.Sleep(100 * time.Millisecond)
	result := id * 10
	results <- result
	fmt.Printf("Worker %d finished\n", id)
}

func main() {
	fmt.Println("Multi-thread program started")

	// Short delay for debugger
	time.Sleep(100 * time.Millisecond)

	const numWorkers = 3
	var wg sync.WaitGroup
	results := make(chan int, numWorkers)

	// Start multiple goroutines
	for i := 1; i <= numWorkers; i++ {
		wg.Add(1)
		go worker(i, &wg, results)
	}

	// Wait for all workers in a separate goroutine
	go func() {
		wg.Wait()
		close(results)
	}()

	// Collect results
	total := 0
	for result := range results {
		total += result
	}

	fmt.Printf("Total: %d\n", total)
	fmt.Println("Multi-thread program completed")

	// Keep running
	time.Sleep(60 * time.Second)
}
